### PR TITLE
Docstrings and formatting improvements

### PR DESCRIPTION
**Pull request by Hasenn**
### **Describe your change:**
I've been correcting some spelling in docstrings and applying PEP8 autoformatting to scripts i read. I didn't make a single PR per script as the review should be pretty straightforward from reading the diffs, and i didn't change much per file.

* [ ] Add an algorithm?
* [x] Fix a bug or typo in an existing algorithm?
* [x] Documentation change?

### **Checklist:**
* [x] I have read [CONTRIBUTING.md](https://github.com/TheAlgorithms/Python/blob/master/CONTRIBUTING.md).
* [x] This pull request is all my own work -- I have not plagiarized.
* [x] I know that pull requests will not be merged if they fail the automated tests.
* [ ] This PR only changes one algorithm file.  To ease review, please open separate PRs for separate algorithms.
* [x] All new Python files are placed inside an existing directory.
* [x] All filenames are in all lowercase characters with no spaces or dashes.
* [x] All functions and variable names follow Python naming conventions.
* [x] All function parameters and return values are annotated with Python [type hints](https://docs.python.org/3/library/typing.html).
* [x] All functions have [doctests](https://docs.python.org/3/library/doctest.html) that pass the automated testing.
* [x] All new algorithms have a URL in its comments that points to Wikipedia or other similar explanation.
* [x] If this pull request resolves one or more open issues then the commit message contains `Fixes: #{$ISSUE_NO}`.

### CODE REVIEW

**Comment by spamegg1** [comment]
Build log says:
> =================================== FAILURES ===================================
_______________________ [doctest] rat_in_maze.solve_maze _______________________
040     >>> solve_maze(maze)
041     [1, 1, 1]
042     [0, 0, 1]
043     [0, 0, 1]
044     True
045 
046     >>> maze = [[0, 1, 0],
047     ...         [0, 1, 0],
048     ...         [1, 0, 0]]
049     >>> solve_maze(maze)
Expected:
    Solution does not exist!
    False
Got:
    Solution does not exists!
    False
/home/travis/build/TheAlgorithms/Python/backtracking/rat_in_maze.py:49: DocTestFailure

**Comment by Hasenn** [comment]
oh, yeah it was a doctest i need to find the print statement too and update it

**Review by poyea — APPROVED** [review]
Thank you for your pull request!🤩

File: boolean_algebra/quine_mc_cluskey.py [hunk 1]
```diff
@@ -131,7 +131,8 @@ def prime_implicant_chart(prime_implicants, binary):
     >>> prime_implicant_chart(['0.00.01.5'],['0.00.01.5'])
     [[1]]
     """
-    chart = [[0 for x in range(len(binary))] for x in range(len(prime_implicants))]
+    chart = [[0 for x in range(len(binary))]
```
Comment (cclauss) [inline boolean_algebra/quine_mc_cluskey.py:134]: My sense is that psf/black will undo this change.

File: cellular_automata/one_dimensional.py [hunk 2]
```diff
@@ -32,10 +32,14 @@ def new_generation(cells: List[List[int]], rule: List[int], time: int) -> List[i
...
     for i in range(population):
         # Get the neighbors of each cell
-        left_neighbor = 0 if i == 0 else cells[time][i - 1]  # special: leftmost cell
-        right_neighbor = 0 if i == population - 1 else cells[time][i + 1]  # rightmost
+        # Handle neighbours outside bounds by using 0 as their value
+        left_neighbor = 0 if i == 0 else cells[time][i - 1]
+        right_neighbor = 0 if i == population - 1 else cells[time][i + 1]
         # Define a new cell and add it to the new generation
-        situation = 7 - int(f"{left_neighbor}{cells[time][i]}{right_neighbor}", 2)
+        situation = 7 - int(
```
Comment (cclauss) [inline cellular_automata/one_dimensional.py:39]: My sense is that psf/black will undo this change.

File: backtracking/rat_in_maze.py [hunk 3]
```diff
@@ -47,13 +47,13 @@ def solve_maze(maze: list) -> bool:
...
     >>> solve_maze(maze)
-    Solution does not exists!
+    Solution does not exist!
     False
 
     >>> maze = [[0, 1],
     ...         [1, 0]]
     >>> solve_maze(maze)
-    Solution does not exists!
+    Solution does not exist!
```
Comment (cclauss) [inline backtracking/rat_in_maze.py:56]:
```suggestion
    No solution exists!
```

File: backtracking/rat_in_maze.py [hunk 4]
```diff
@@ -63,16 +63,16 @@ def solve_maze(maze: list) -> bool:
     if solved:
         print("\n".join(str(row) for row in solutions))
     else:
-        print("Solution does not exists!")
+        print("Solution does not exist!")
```
Comment (cclauss) [inline backtracking/rat_in_maze.py:66]:
```suggestion
        print("No solution exists!")
```

Comment (cclauss) [reply backtracking/rat_in_maze.py:66]: I would raise an exception but that is a personal choice.

**Comment by Hasenn** [comment]
thanks @cclauss i applied black and changed the text to "No solution exists". I don't want to raise an exception to avoid having long error messages in the doctests

**Review by cclauss — APPROVED** [review]

**Review by shellhub — APPROVED** [review]
